mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/operation/OperationResult.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.operation;
+
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.api.metadata.MediaType;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Represents the result of an operation execution. Extensions can use this class
+ * for cases in which the operation not only needs to return a value to be set
+ * on the message payload but also wants to specify message attributes and/or
+ * {@link MediaType}.
+ * <p>
+ * The {@link #getOutput()} value is always taken at face value, meaning that
```

Comment:
We need to define a mechanism for operations to have return types. An operation can return "void" which means it's not supposed to affect the message at all. In this cases target attribute cannot be used in the operation. Or an operation can return an OUTPUT. Then it's up to the runtime to decide whether the message needs to be replaced with the new one or just send the same message it was used to invoke the operation, to the next message processor.
Can you create an issue to track this requirement?